java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/explorer/TreeNodeRegistry.java

```diff
@@ -40,6 +42,19 @@
     CompletionStage<TreeViewProvider> createProvider(String id);
     
     public class ImageDataOrIndex {
+        /**
+         * Base URL of the image, if it could be extracted from the Image object.
+         */
+        public URI          baseURI;
+        
+        /**
+         * URLs(?) of additional image constituents.
```

Comment:
Here is some javadoc for `composition`. Surprises me there is `IconDescriptor` and `ImageDataOrIndex`...

Reply:
I know - the protocol need to be cleared up, ideally during NB13 release cycle. The client should call back with some getImageData(). I do not want to mess with existing parts because of the vscode release timing, but I am planning to clean it up so the LSP server releases clean with NB13.